### PR TITLE
Improve the probe dictionary C++ guard display #445

### PR DESCRIPTION
**Pull request by shreya99oak**

### CODE REVIEW

File: probe-dictionary/explorer.css [hunk 1]
```diff
@@ -122,3 +122,8 @@ body {
 .hidden {
     display: none;
 }
+
+.grow:empty:after{
```
Comment (chutten) [inline probe-dictionary/explorer.css]: `.grow` is a very generic class to be hinging this behaviour on. I think `#detail-cpp-guard` may be the better selector for this (though we lose out on getting this behaviour for bug numbers unless we add it specifically)

**Review by chutten — CHANGES_REQUESTED** [review]

File: probe-dictionary/explorer.css (hunk 1 above)
Comment (shreya99oak) [reply probe-dictionary/explorer.css]: Ok, I'll add the id instead...should I add the id for bug numbers too?

**Comment by georgf** [comment]
This addresses #445.

**Comment by georgf** [comment]
Can you provide [a live-link to the changes](https://github.com/mozilla/telemetry-dashboard/#submitting-pull-requests-for-the-dashboard) so we can more easily review this?
Thanks!

**Comment by shreya99oak** [comment]
Hi,
https://shreya99oak.github.io/probe-dictionary/index.html
I have attached a screenshot too
changes
..

**Comment by georgf** [comment]
@shreya99oak, the live link doesn't load for me currently.
Looking in the web console i see some errors:
screen shot 2018-03-08 at 19 53 45

**Comment by shreya99oak** [comment]
Hi, pls. try this link https://shreya99oak.github.io/telemetry-dashboard-probe-dictionary/probe-dictionary/?search=&searchtype=in_any&optout=false&channel=any&constraint=is_in&version=any&view=search-results-view

**Comment by shreya99oak** [comment]
@georgf  Pls. review the changes made for #456 too https://shreya99oak.github.io/telemetry-dashboard-probe-dictionary/

File: _config.yml [hunk 2]
```diff
@@ -0,0 +1 @@
+
```
Comment (georgf) [inline _config.yml]: This should not be in the PR, please take it out.

File: index.html [hunk 3]
```diff
@@ -5,6 +5,18 @@
     <meta http-equiv="X-UA-Compatible" content="IE=edge">
     <meta name="viewport" content="width=device-width, initial-scale=1">
     <title>Telemetry portal</title>
+    <style>
```
Comment (georgf) [inline index.html]: There is no need to add CSS here, let's move this into explorer.css.

File: index.html [hunk 4]
```diff
@@ -5,6 +5,18 @@
     <meta http-equiv="X-UA-Compatible" content="IE=edge">
     <meta name="viewport" content="width=device-width, initial-scale=1">
     <title>Telemetry portal</title>
+    <style>    
+    .contrast:after{
```
Comment (georgf) [inline index.html]: For coding style, please put a space before the bracket.

File: index.html [hunk 5]
```diff
@@ -5,6 +5,18 @@
...
+    content:"";
```
Comment (georgf) [inline index.html]: Similarly, we use a space after the `:`.
Similar below.

File: index.html [hunk 6]
```diff
@@ -5,6 +5,18 @@
...
+    position:absolute;
+    height:30px;
+    background-color:#DEDEDE;
+    padding: 35px ;
+    top:0;
+    left:0;
+    right:0;
+    z-index:-1;  }
```
Comment (georgf) [inline index.html]: The closing bracket should be on a new line.

File: probe-dictionary/explorer.css [hunk 7]
```diff
@@ -122,3 +122,8 @@ body {
 .hidden {
     display: none;
 }
+
+#detail-cpp-guard:empty:after,#detail-bug-numbers:empty:after{
```
Comment (georgf) [inline probe-dictionary/explorer.css]: Here we use spaces after the `,` and before the `{`.

**Review by georgf — COMMENTED** [review]
Visually this looks good!
I left some style comments below that we need to fix, but nothing major.

File: _config.yml (hunk 2 above)
Comment (shreya99oak) [reply _config.yml]: Yes, sorry I'll take that out

File: probe-dictionary/explorer.css (hunk 7 above)
Comment (shreya99oak) [reply probe-dictionary/explorer.css]: I'll make the changes:) Thanks!

**Comment by shreya99oak** [comment]
Hi, How do I remove d02ab77 from the PR?

**Comment by chutten** [comment]
@shreya99oak You might find `git rebase -i` to be helpful for removing and reordering commits in your local tree.

If you rewrite history in this way, you'll need to force-push with `-f` to your `gh-pages` branch to update this PR

**Comment by shreya99oak** [comment]
I don't think I was able to rebase...I typed git rebase HEAD~9 , then deleted the config.yml lines on the text editor..but in vain

**Comment by shreya99oak** [comment]
Somehow the other commits have appeared too, i don't understand why..pls. help

**Comment by georgf** [comment]
Did you put both changes on the same branch?
One way to work on more than issue at once is to use different branches for each (see [here](https://help.github.com/articles/about-branches/) and [here](https://help.github.com/articles/creating-and-deleting-branches-within-your-repository/)).

If it's too hard to restore the right state with commands, you could always just start new branches and copy the changes you want in.

**Comment by shreya99oak** [comment]
Hi, I changed the branch and worked on  the issue #449, then created a new PR, still somehow the earlier commits appear...

**Comment by shreya99oak** [comment]
Should I clone the main repo( instead of my fork) and then create a branch,make the changes and push it? Would that work?

**Comment by georgf** [comment]
The standard workflow is to work on your own fork.
If you are just working on one issue at a time, you can just directly commit to the main branch on it, then open a pull request on this repository.

If you work on more than one issue at a time, you can create a branch for each of them (e.g. one called "improve-cpp-guard". Then you switch to that branch whenever you work on that issue.

I found the [Github desktop application](https://desktop.github.com/) convenient to work with this, which e.g. has an "add branch" button and let's you just select which branch you want to work on currently. After installing it, in your forked the repository, you can go to "clone or download", then choose "open in desktop".

**Pull request closed without merging**